### PR TITLE
Unit test added using Mocha and Chai

### PR DESCRIPTION
**Pull request by jayserdny**
**Changes made**

Unit test were added using Mocha and Chai. It helps contributors and project owner to easily test the functions if some changes are made. It ensure the functionality of the functions.

Here is an example if you run the testrunner.html file:

screen shot 2018-02-10 at 4 20 17 pm

Library code was not touched. However, I ran the grunt task again just in case.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -24,5 +24,5 @@ module.exports = function(grunt) {
 	});
 	grunt.loadNpmTasks('grunt-babel');
 	grunt.loadNpmTasks('grunt-contrib-uglify');
-	grunt.registerTask("default", ["babel","uglify"]);
+	grunt.registerTask('default', ['babel', 'uglify']);
```
Comment (jayserdny) [inline Gruntfile.js]: Not really necessary but it is a good practice in javascript to use single quote for strings

File: package.json [hunk 2]
```diff
@@ -25,7 +25,9 @@
   "devDependencies": {
     "babel-core": "^6.26.0",
     "babel-preset-env": "^1.6.1",
+    "chai": "^4.1.2",
```
Comment (jayserdny) [inline package.json]: This is one of the dependencies needed for the unit test

File: package.json [hunk 3]
```diff
@@ -25,7 +25,9 @@
...
     "grunt-babel": "^7.0.0",
-    "grunt-contrib-uglify": "^3.3.0"
+    "grunt-contrib-uglify": "^3.3.0",
+    "mocha": "^5.0.0"
```
Comment (jayserdny) [inline package.json]: and here is the another dependence to run unit test

File: test/spec.utopian.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ * 
+ * @author Jayser Mendez
+ * @version 0.0.1
+ */
+
+'use strict';
+var assert = chai.assert;
+var expect = chai.expect;
+
+describe('Function to get moderators', function () {
```
Comment (jayserdny) [inline test/spec.utopian.js]: This specification is in charge to ensure that the function to get moderators is returning an expected output

File: test/spec.utopian.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      it('should return an object with an array called result which holds objects or empty array', function () {
+
+            return utopian.getModerators()
+                  .then(function(moderators) {
+                        expect(moderators).to.be.a('object');
```
Comment (jayserdny) [inline test/spec.utopian.js]: The response itself is expected to be an object

File: test/spec.utopian.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                        expect(moderators.results).to.be.a('array');
```
Comment (jayserdny) [inline test/spec.utopian.js]: And the attribute results of the moderators object is expected to be an array

File: test/spec.utopian.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+                        expect(moderators.total).to.be.a('number');
```
Comment (jayserdny) [inline test/spec.utopian.js]: While the attribute total is expected to be a number

File: testrunner.html [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    <title>Mocha Tests</title>
+    <link rel="stylesheet" href="node_modules/mocha/mocha.css">
+  </head>
+  <body>
+    <div id="mocha"></div>
+    <script src="node_modules/mocha/mocha.js"></script>
+    <script src="node_modules/chai/chai.js"></script>
+    <script src="http://code.jquery.com/jquery-3.3.1.min.js"></script>
+    <script src="dist/utopian.min.js"></script>
+    <script>mocha.setup('bdd')</script>
```
Comment (jayserdny) [inline testrunner.html]: This line is in change of starting the bootstrap of the mocha to be able to run unit test

File: testrunner.html [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+    <script src="test/spec.utopian.js"></script>
```
Comment (jayserdny) [inline testrunner.html]: the spec.utopian.js file is loaded after initializing the mocha unit test and before running it of course

File: testrunner.html [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+    <script>
+      mocha.run();
```
Comment (jayserdny) [inline testrunner.html]: As simple as it says, run the unit test

**Comment by ms10398** [comment]
That was a well detailed PR with perfect comments.

I will run the tests locally and merge the PR just as I complete it.

It looks good to me in the first look.

**Comment by jayserdny** [comment]
@ms10398 It can looks good but test is needed to ensure its functionality :) We have not compilers in our head :P

**Comment by ms10398** [comment]
Yes thats why it is not merged yet

I am just checking it and then will merge your PR.

**Comment by jayserdny** [comment]
Sure :) I will design a chip to adapt a compiler to our brain :D Just kidding

**Comment by ms10398** [comment]
I will add more functionalities to it soon

maybe just as I wake up tomorrow as its already 4AM.

and yes with your mocha tests surely! :smile: 

Thanks for the contribution.

**Comment by jayserdny** [comment]
Oh man, go to sleep hahahahah. And you are welcome :)

**Pull request merged**